sdk/iot/hub/src/az_iot_hub_client_twin.c

```diff
@@ -121,7 +121,11 @@ AZ_NODISCARD az_result az_iot_hub_client_twin_document_get_publish_topic(
   remainder = az_span_copy_u8(remainder, az_iot_hub_client_twin_question);
   remainder = az_span_copy(remainder, az_iot_hub_client_request_id_span);
   remainder = az_span_copy_u8(remainder, az_iot_hub_client_twin_equals);
-  remainder = az_span_copy(remainder, request_id);
+
+  // Ignore the result value since we have checked that there is enough space
+  az_result unused = az_span_u32toa(remainder, request_id, &remainder);
+  (void)unused;
```

Comment:
nit: you can also use AZ_RETURN_IF_FAILED(...)  instead of void-ing the result.
That way, if this ever fail in the future, you would be returning instead of letting it go

Reply:
That's what was there originally, @vhvb1989 - it was an unnecessary check.